src/beast/beast/asio/streambuf.h

```diff
@@ -442,6 +442,7 @@ basic_streambuf<Allocator>::prepare (size_type n) ->
                     ++pos;
                     break;
                 }
+                out_end_ = pos->size();
```

Comment:
This is the best I could come up with. I have no problem with what's there though.
```
                auto const last_n = n;
                n -= std::min (n, pos->size ());
                if (n == 0)
                {
                    out_end_ = last_n;
                    ++pos;
                    break;
                }
```